apps/new-dealer/controllers/storage-address.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const BaseController = require('hof').controllers.base;
+const _ = require('lodash');
+
+module.exports = class AddressController extends BaseController {
+  locals(req, res, callback) {
+    const isManual = req.params.action === 'manual';
+    const locals = super.locals(req, res, callback);
+    const postcode = req.sessionModel.get('storage-postcode');
+    const addresses = req.sessionModel.get('storageAddresses');
+    const hasStorageAddresses = req.sessionModel.get('storageAddresses') ? true : false;
+    const storageAddresses = [];
+    _.forEach(addresses, (value, key) => {
+      const address = {
+        id: key,
+        address: value.address
+      };
+      storageAddresses.push(address);
+    });
+    return Object.assign({}, locals, {
+      storageAddresses,
+      hasStorageAddresses,
+      postcode,
+      postcodeApiMessageKey: isManual ? '' : (req.sessionModel.get('postcodeApiMeta') || {}).messageKey
+    });
+  }
+
+  getBackLink(req, res, callback) {
+    const backLink = super.getBackLink(req, res, callback);
+    if (req.params.id && req.params.id !== undefined) {
+      const id = req.params.id;
+      const back = req.params.action === 'edit' ? `${backLink}/${id}` : `${backLink}/edit/${id}`;
+      return back;
+    }
+    return backLink;
+  }
+
+  getValues(req, res, callback) {
+    super.getValues(req, res, (err, values) => {
+      if (err) {
+        // eslint-disable-next-line no-console
+        callback(err);
+        return callback(null, values);
```

Comment:
why are we calling callback twice? If there's an error we want to return callback(err) - dont worry about passing it values. Pretty sure this will actually break as headers are already set